packages/trpc/server/routers/viewer/slots/util.ts

```diff
@@ -200,6 +201,19 @@ export async function getSchedule(input: TGetScheduleInputSchema) {
     logger.setSettings({ minLevel: "silly" });
   }
   const startPrismaEventTypeGet = performance.now();
+
+  if (!input.eventTypeId && input.eventTypeSlug && !!input.usernameList) {
```

Comment:
I think we shouldn't get eventTypeId here. The Booker already had the eventTypeId and it should pass the info to getSchedule.

Even if we need to do this, let's abstract it out in a function like getEventTypeId

Reply:
actually the booker get the eventTypeId from trpc.event.public so slots.getSchedule would only run after trpc.event.public ( in useEvent) return us result but here i created a function getPublicEventId which does what trpc.public.event ( getPublicEvent) does but fetches nothing extra only id so it should be a bit faster and we don't wait for trpc.public.event api to finish

Reply:
public.event is prefetched on server, so on client it should be available on first render. There was a bug in that, which I fixed in my PR